### PR TITLE
[fix]: [cds-59179]: Log warning to user when a non-null attribute in manifest is null or has value of runtime input expression

### PR DESCRIPTION
**Pull request by lovish1999**
## Describe your changes
This bug occurred because the user modified the manifest identifier from yaml during runtime execution of pipeline. This resulted in the manifests not getting merged properly and the value for runtime inputs remained as <+input>. Now, to handle this, we will display a warning to the user in case the value of any required field of the service manifest is null or of form "<+input>".
Screenshot 2023-05-09 at 7 39 39 PM

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/47410)

### CODE REVIEW

**Comment by lovish1999** [comment]
trigger uts

File: 125-cd-nextgen/src/main/java/io/harness/cdng/plugininfoproviders/AwsSamBuildPluginInfoProvider.java [hunk 1]
```diff
@@ -46,10 +42,10 @@
 import com.google.inject.Inject;
 import com.google.inject.name.Named;
 import java.io.IOException;
-import java.util.*;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
```
Comment (lovish1999) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/plugininfoproviders/AwsSamBuildPluginInfoProvider.java:48]: This file is unrelated to my changes. It was failing code format check, so fixed it.

File: 125-cd-nextgen/src/main/java/io/harness/cdng/plugininfoproviders/AwsSamDeployPluginInfoProvider.java [hunk 2]
```diff
@@ -25,8 +24,10 @@
 import io.harness.connector.ConnectorResponseDTO;
 import io.harness.connector.services.ConnectorService;
 import io.harness.delegate.beans.connector.ConnectorConfigDTO;
-import io.harness.delegate.beans.connector.awsconnector.*;
-import io.harness.exception.InvalidRequestException;
+import io.harness.delegate.beans.connector.awsconnector.AwsConnectorDTO;
+import io.harness.delegate.beans.connector.awsconnector.AwsCredentialDTO;
+import io.harness.delegate.beans.connector.awsconnector.AwsCredentialSpecDTO;
+import io.harness.delegate.beans.connector.awsconnector.AwsManualConfigSpecDTO;
```
Comment (lovish1999) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/plugininfoproviders/AwsSamDeployPluginInfoProvider.java:30]: This file is unrelated to my changes. It was failing code format check, so fixed it.

File: 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasStepHelper.java [hunk 3]
```diff
@@ -131,7 +130,6 @@
 import io.harness.delegate.task.pcf.artifact.AwsS3TasArtifactRequestDetails;
 import io.harness.delegate.task.pcf.artifact.AzureDevOpsTasArtifactRequestDetails;
 import io.harness.delegate.task.pcf.artifact.BambooTasArtifactRequestDetails;
-import io.harness.delegate.task.pcf.artifact.CustomArtifactTasRequestDetails;
```
Comment (lovish1999) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/tas/TasStepHelper.java:134]: This file is unrelated to my changes. It was failing code format check, so fixed it.

File: 125-cd-nextgen/src/test/java/io/harness/cdng/tas/TasStepHelperTest.java [hunk 4]
```diff
@@ -176,7 +176,6 @@
 import io.harness.delegate.task.pcf.artifact.AwsS3TasArtifactRequestDetails;
 import io.harness.delegate.task.pcf.artifact.AzureDevOpsTasArtifactRequestDetails;
 import io.harness.delegate.task.pcf.artifact.BambooTasArtifactRequestDetails;
-import io.harness.delegate.task.pcf.artifact.CustomArtifactTasRequestDetails;
```
Comment (lovish1999) [inline 125-cd-nextgen/src/test/java/io/harness/cdng/tas/TasStepHelperTest.java:179]: This file is unrelated to my changes. It was failing code format check, so fixed it.

**Comment by yogesh-chauhan** [comment]
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/GitStore.java` line 158 at r2](https://reviewable.io/reviews/harness/harness-core/47410#-NUvTOa2-0zmht7XoHFO:-NUvTOa2-0zmht7XoHFP:b-t1sjcg) ([raw file](https://github.com/harness/harness-core/blob/710d20cae2e36537da8e30df4cd18e5b8aa45c3b/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/GitStore.java#L158)):*
> ```Java
>   @Override
>   public void validate(NGLogCallback logCallback, String manifestIdentifier) {
>     StoreConfigHelper.validateGitStoreType(
> ```

validate should just validate and not log

**Comment by lovish1999** [comment]
trigger uts

**Comment by yogesh-chauhan** [comment]
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/configfile/validator/IndividualConfigFileStepValidator.java` line 52 at r3](https://reviewable.io/reviews/harness/harness-core/47410#-NV-6nZ305Veu5AVNW40:-NV-6nZ48TA7IJf1D0WN:b-vwfc6) ([raw file](https://github.com/harness/harness-core/blob/f9f79a2e2d41364b179261764dc11eeeef052111/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/configfile/validator/IndividualConfigFileStepValidator.java#L52)):*
> ```Java
>       validateHarnessStore(configFileIdentifier, (HarnessStore) store, allowExpression);
>     }
>     return invalidParameters;
> ```

Lets do this
```
return store.validateAtRuntime();
```

**Comment by yogesh-chauhan** [comment]
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java` line 83 at r3](https://reviewable.io/reviews/harness/harness-core/47410#-NV-7Yc40IKx4XpQi9dI:-NV-7Yc40IKx4XpQi9dJ:b-ukycgu) ([raw file](https://github.com/harness/harness-core/blob/f9f79a2e2d41364b179261764dc11eeeef052111/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java#L83)):*
> ```Java
> 
>     return invalidParameters;
>   }
> ```

Please write unit tests for this

**Comment by yogesh-chauhan** [comment]
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/configfile/steps/ConfigFilesStepV2.java` line 156 at r3](https://reviewable.io/reviews/harness/harness-core/47410#-NV-7p3_51WBLjIvSDgM:-NV-7p3_51WBLjIvSDgN:bqxqyu6) ([raw file](https://github.com/harness/harness-core/blob/f9f79a2e2d41364b179261764dc11eeeef052111/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/configfile/steps/ConfigFilesStepV2.java#L156)):*
> ```Java
>       Set<String> invalidParameters =
>           IndividualConfigFileStepValidator.validateConfigFileAttributes(identifier, spec, true);
>       if (!invalidParameters.isEmpty()) {
> ```

use `EmptyPredicate.isEmpty()`

**Comment by yogesh-chauhan** [comment]
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/configfile/steps/ConfigFilesStepV2.java` line 163 at r3](https://reviewable.io/reviews/harness/harness-core/47410#-NV-8AzF44M1rZYaxDb7:-NV-8AzF44M1rZYaxDb8:b-awemhm) ([raw file](https://github.com/harness/harness-core/blob/f9f79a2e2d41364b179261764dc11eeeef052111/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/configfile/steps/ConfigFilesStepV2.java#L163)):*
> ```Java
>             LogLevel.WARN);
>         invalidParameters.forEach(
>             invalidParameter -> { logCallback.saveExecutionLog(format("%s, ", invalidParameter), LogLevel.WARN); });
> ```

Please post a screenshot of how this is looking on the UI

**Comment by yogesh-chauhan** [comment]
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/configfile/steps/ConfigFilesStepV2.java` line 207 at r3](https://reviewable.io/reviews/harness/harness-core/47410#-NV-8Hem7dEO2lbJYBB9:-NV-8Hem7dEO2lbJYBBA:b-k6pn89) ([raw file](https://github.com/harness/harness-core/blob/f9f79a2e2d41364b179261764dc11eeeef052111/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/configfile/steps/ConfigFilesStepV2.java#L207)):*
> ```Java
>             LogLevel.WARN);
>         invalidParameters.forEach(
>             invalidParameter -> { logCallback.saveExecutionLog(format("%s, ", invalidParameter), LogLevel.WARN); });
> ```

duplicate code

**Comment by yogesh-chauhan** [comment]
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/configfile/steps/IndividualConfigFileStep.java` line 67 at r3](https://reviewable.io/reviews/harness/harness-core/47410#-NV-8Q232LhCtB0TMIQA:-NV-8Q232LhCtB0TMIQB:bbamny4) ([raw file](https://github.com/harness/harness-core/blob/f9f79a2e2d41364b179261764dc11eeeef052111/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/configfile/steps/IndividualConfigFileStep.java#L67)):*
> ```Java
>       invalidParameters.forEach(
>           invalidParameter -> { logCallback.saveExecutionLog(format("%s, ", invalidParameter), LogLevel.WARN); });
>     }
> ```

lets not do the change in service v1 if the bug is not from service v1

**Comment by yogesh-chauhan** [comment]
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java` line 137 at r3](https://reviewable.io/reviews/harness/harness-core/47410#-NV-8xj_D0-OVxXMvVZ4:-NV-8xj_D0-OVxXMvVZ5:bu6py4f) ([raw file](https://github.com/harness/harness-core/blob/f9f79a2e2d41364b179261764dc11eeeef052111/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java#L137)):*
> ```Java
>     manifestAttributes.forEach(manifestAttribute -> {
>       Set<String> invalidParameters = manifestAttribute.validateAtRuntime();
>       if (!invalidParameters.isEmpty()) {
> ```

empty predicate

**Comment by yogesh-chauhan** [comment]
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/yaml/S3UrlStoreConfig.java` line 128 at r3](https://reviewable.io/reviews/harness/harness-core/47410#-NV-9J3S7MEZhi2nF-Cj:-NV-9J3S7MEZhi2nF-Ck:bv9x6zs) ([raw file](https://github.com/harness/harness-core/blob/f9f79a2e2d41364b179261764dc11eeeef052111/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/yaml/S3UrlStoreConfig.java#L128)):*
> ```Java
>     }
>     if (StoreConfigHelper.checkStringParameterNullOrInput(region)) {
>       invalidParameters.add("region");
> ```

use @FieldNameConstants(innerTypeName = "S3UrlStoreConfigKeys" ) and similar in all the other java classes

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 4 of 25 files at r2, 1 of 23 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47410)* status: 22 file(s) to review. 0 of 1 approvals obtained.

**Comment by lovish1999** [comment]
trigger uts

**Comment by lovish1999** [comment]
trigger messagecheck

**Comment by yogesh-chauhan** [comment]
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/ManifestAttributes.java` line 38 at r4](https://reviewable.io/reviews/harness/harness-core/47410#-NV8ZMukD4-XwEBUDLJQ:-NV8ZMukD4-XwEBUDLJR:b-b10mee) ([raw file](https://github.com/harness/harness-core/blob/08216557af869d3b92fd37667beae7c63cb211ae/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/ManifestAttributes.java#L38)):*
> ```Java
> 
>   interface ManifestAttributeStepParameters {}
>   default Set<String> validateAtRuntime() {
> ```

store config can be null. so lets check for that

**Comment by yogesh-chauhan** [comment]
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/S3StoreConfig.java` line 146 at r4](https://reviewable.io/reviews/harness/harness-core/47410#-NV8ZYYz8wi5MK_1Ztoe:-NV8ZYYz8wi5MK_1Ztof:bcw53aw) ([raw file](https://github.com/harness/harness-core/blob/08216557af869d3b92fd37667beae7c63cb211ae/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/S3StoreConfig.java#L146)):*
> ```Java
>       invalidParameters.add(S3StoreConfigKeys.bucketName);
>     }
>     if (StoreConfigHelper.checkStringParameterNullOrInput(region)) {
> ```

is region a mandatory field ?

**Comment by yogesh-chauhan** [comment]
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java` line 24 at r4](https://reviewable.io/reviews/harness/harness-core/47410#-NV8ZrvE9VBmfSzdwFfT:-NV8ZrvE9VBmfSzdwFfU:b-kmwu60) ([raw file](https://github.com/harness/harness-core/blob/08216557af869d3b92fd37667beae7c63cb211ae/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java#L24)):*
> ```Java
> @UtilityClass
> public class StoreConfigHelper {
>   public boolean checkListOfStringsParameterNullOrInput(ParameterField<List<String>> parameterField) {
> ```

- put a java doc for this method
- also maybe a better name for it

**Comment by yogesh-chauhan** [comment]
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java` line 33 at r4](https://reviewable.io/reviews/harness/harness-core/47410#-NV8Zy7i4yeJAVJRhqch:-NV8Zy7i4yeJAVJRhqci:bihf4wa) ([raw file](https://github.com/harness/harness-core/blob/08216557af869d3b92fd37667beae7c63cb211ae/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java#L33)):*
> ```Java
>         return true;
>       }
>     } else {
> ```

can we check for isInstanceOf(List) here ?

**Comment by yogesh-chauhan** [comment]
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java` line 81 at r4](https://reviewable.io/reviews/harness/harness-core/47410#-NV8_593-10e4VNkE_Yp:-NV8_593-10e4VNkE_Yq:b-tx0hh6) ([raw file](https://github.com/harness/harness-core/blob/08216557af869d3b92fd37667beae7c63cb211ae/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java#L81)):*
> ```Java
>     if (StoreConfigHelper.checkStringParameterNullOrInput(folderPath)
>         && StoreConfigHelper.checkListOfStringsParameterNullOrInput(paths)) {
>       invalidParameters.add("folderPath");
> ```

extract these out as constants in this same class itself

**Comment by yogesh-chauhan** [comment]
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java` line 62 at r4](https://reviewable.io/reviews/harness/harness-core/47410#-NV8_A03C4mf8EHf8erq:-NV8_A03C4mf8EHf8err:b77u7sy) ([raw file](https://github.com/harness/harness-core/blob/08216557af869d3b92fd37667beae7c63cb211ae/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java#L62)):*
> ```Java
>       ParameterField<List<String>> paths, ParameterField<String> branch, ParameterField<String> commitId,
>       FetchType gitFetchType) {
>     Set<String> invalidParameters = new HashSet<>();
> ```

nit: can be a final field

**Comment by yogesh-chauhan** [comment]
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java` line 68 at r4](https://reviewable.io/reviews/harness/harness-core/47410#-NV8_DVtEvgKV_cC_ys7:-NV8_DVu6UjfI5b5kstZ:b-oovhsx) ([raw file](https://github.com/harness/harness-core/blob/08216557af869d3b92fd37667beae7c63cb211ae/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java#L68)):*
> ```Java
>     }
>     if (gitFetchType != null) {
>       if (gitFetchType.equals(FetchType.BRANCH)) {
> ```

constant should be on the left side of the comparison. You don't need a null check in that case

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 6 of 23 files at r3, 21 of 23 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47410)* status: 3 file(s) to review. 0 of 1 approvals obtained.

**Review by lovish1999 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47410)* status: 3 file(s) to review. 6. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java` line 137 at r3](https://reviewable.io/reviews/harness/harness-core/47410#-NV-8xj_D0-OVxXMvVZ4:-NV9Xc0qD3VzNiMN8TeH:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/f9f79a2e2d41364b179261764dc11eeeef052111/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/manifest/steps/ManifestsStepV2.java#L137)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

empty predicate
</blockquote></details>

Done.
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/ManifestAttributes.java` line 38 at r4](https://reviewable.io/reviews/harness/harness-core/47410#-NV8ZMukD4-XwEBUDLJQ:-NV9XdLj64N6OcW80bJv:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/08216557af869d3b92fd37667beae7c63cb211ae/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/ManifestAttributes.java#L38)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

store config can be null. so lets check for that 
</blockquote></details>

Done.
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/S3StoreConfig.java` line 146 at r4](https://reviewable.io/reviews/harness/harness-core/47410#-NV8ZYYz8wi5MK_1Ztoe:-NV9Xe5p9XJsbJcK6A7n:b-8s7x6c) ([raw file](https://github.com/harness/harness-core/blob/08216557af869d3b92fd37667beae7c63cb211ae/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/S3StoreConfig.java#L146)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

is region a mandatory field ?
</blockquote></details>

yes

___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java` line 83 at r3](https://reviewable.io/reviews/harness/harness-core/47410#-NV-7Yc40IKx4XpQi9dI:-NV9Zioa1nOsPnKN4shp:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/f9f79a2e2d41364b179261764dc11eeeef052111/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java#L83)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

Please write unit tests for this 
</blockquote></details>

Done.
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java` line 68 at r4](https://reviewable.io/reviews/harness/harness-core/47410#-NV8_DVtEvgKV_cC_ys7:-NV9Zbhi8pAdyrvsn_cV:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/08216557af869d3b92fd37667beae7c63cb211ae/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java#L68)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

constant should be on the left side of the comparison. You don't need a null check in that case 
</blockquote></details>

Done.
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java` line 81 at r4](https://reviewable.io/reviews/harness/harness-core/47410#-NV8_593-10e4VNkE_Yp:-NV9ZcIZDZ4jR08oKX7R:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/08216557af869d3b92fd37667beae7c63cb211ae/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java#L81)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

extract these out as constants in this same class itself
</blockquote></details>

Done.

**Review by lovish1999 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47410)* status: 3 file(s) to review. 8. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java` line 24 at r4](https://reviewable.io/reviews/harness/harness-core/47410#-NV8ZrvE9VBmfSzdwFfT:-NV9aymn45LEIjZ2_P5w:b32w1t) ([raw file](https://github.com/harness/harness-core/blob/08216557af869d3b92fd37667beae7c63cb211ae/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java#L24)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

- put a java doc for this method
- also maybe a better name for it
</blockquote></details>

Ok
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java` line 33 at r4](https://reviewable.io/reviews/harness/harness-core/47410#-NV8Zy7i4yeJAVJRhqch:-NV9b1-AD-M9QzD7nndt:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/08216557af869d3b92fd37667beae7c63cb211ae/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java#L33)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

can we check for isInstanceOf(List) here ?
</blockquote></details>

Done.
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java` line 62 at r4](https://reviewable.io/reviews/harness/harness-core/47410#-NV8_A03C4mf8EHf8erq:-NV9b5-n81q-6W_NtvLk:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/08216557af869d3b92fd37667beae7c63cb211ae/127-cd-nextgen-entities/src/main/java/io/harness/cdng/manifest/yaml/StoreConfigHelper.java#L62)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

nit: can be a final field 
</blockquote></details>

Done

**Comment by lovish1999** [comment]
trigger uts

**Comment by lovish1999** [comment]
trigger uts

**Pull request merged**